File: GameEngine_IOS/GameEngine/Models/SkyBox.m
#import "SkyBox.h"

@implementation SkyBox

/**
 * The initiator of the skyBox
 *
 * @param aTexture
 *            the Identifier of the texture of the sky
 * @param aModel
 *            The model of the sky box
 */
- (id)init : (int) aTextureId : (RawModel*) aModel {
    self = [super init];
    if (self) {
        self -> _textureId = aTextureId;
        self -> _model = aModel;
    }
    return self;
}

@end

File: GameEngine_IOS/GameEngine/Models/SkyBox.swift
import Foundation

/**
* Represents a box with sky textures
*/
public class SkyBox  {
    
    /**
    * RawModel of the skyBox
    */
    var model : RawModel
    
    /**
    * The identifier of the sky box cubic texture
    */
    var textureId : Int
    
    /**
    * The initiator of the skyBox
    *
    * @param aTextureId
    *            the Identifier of the texture of the sky
    * @param aModel
    *            The model of the sky box
    */
    public init(textureId : Int, model : RawModel) {
        self.textureId = textureId;
        self.model = model;
    }
}
